ms2/src/org/labkey/ms2/proteinCoverageMap.jsp

```diff
@@ -24,20 +24,36 @@
 <%@ page import="org.labkey.ms2.MS2Controller" %>
 <%@ page import="java.awt.*" %>
 <%@ page import="java.util.ArrayList" %>
+<%@ page import="java.util.Collections" %>
 <%@ page import="java.util.HashMap" %>
 <%@ page import="java.util.List" %>
 <%@ page import="java.util.Map" %>
 <%@ page import="java.util.Set" %>
+<%@ page import="java.util.TreeMap" %>
 <%@ page import="java.util.TreeSet" %>
 <%@ page import="java.util.stream.Collectors" %>
-<%@ page import="java.util.Collections" %>
+<%@ page import="java.util.Arrays" %>
+<%@ page import="org.labkey.api.protein.PeptideCharacteristic" %>
 <%@ page extends="org.labkey.api.jsp.JspBase" %>
+<%@ taglib prefix="labkey" uri="http://www.labkey.org/taglib" %>
 <%
     MS2Controller.ProteinViewBean bean = ((JspView<MS2Controller.ProteinViewBean>)HttpView.currentView()).getModelBean();
     var currentURL = getActionURL();
     var viewByParam = currentURL.getParameter("viewBy");
+    var replicateIdParam = currentURL.getParameter("replicateId");
+    var peptideFormParam = currentURL.getParameter("peptideForm");
     var isIntensityView = viewByParam == null || viewByParam.equalsIgnoreCase("intensity");
     var isConfidenceView = viewByParam != null && viewByParam.equalsIgnoreCase("confidenceScore");
+    var isCombinedOrStacked = peptideFormParam == null || peptideFormParam.equalsIgnoreCase("combined") ? "combined" : peptideFormParam.equalsIgnoreCase("stacked") ? "stacked" : "combined";

```

Comment:
Can we use a constant for `combined` and `stacked`?

Reply:
And can we switch this to be a Boolean variable. Looks like we force the value to either combined or stacked, so isCombined or isStacked being true/false would be enough and simplify the code a bit.